### PR TITLE
bootstrap flash helper refactor and unit tests

### PR DESCRIPTION
**Pull request by KevinMulhern**
This refactors the bootstrap flash helper and adds a unit test for it.

Fixes:
https://github.com/TheOdinProject/theodinproject/issues/220

### CODE REVIEW

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -497,9 +497,11 @@
   position: relative;
   z-index: 11;
 }
+
 .alert-error{
```
Comment (arun1595) [inline app/assets/stylesheets/static_pages.scss]: Can we get rid of this style?

File: spec/helpers/application_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    context 'when flash type is notice' do
+      let(:flash_type) { 'notice' }
+
+      it 'returns the bootstrap success class' do
+        expect(helper.bootstrap_class_for(flash_type)).to eql('alert-success')
+      end
+    end
+
+    context 'when flash type is error' do
```
Comment (arun1595) [inline spec/helpers/application_helper_spec.rb]: Maybe we can rephrase it to `context 'when custom flash type' do`?

Comment (KevinMulhern) [inline spec/helpers/application_helper_spec.rb]: Yeah thats much better, I'll make that change

File: spec/helpers/application_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      let(:flash_type) { 'error' }
```
Comment (arun1595) [inline spec/helpers/application_helper_spec.rb]: Lets replace `'error'` with `:error` as the key is always going to be a `symbol`

File: spec/helpers/application_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+describe ApplicationHelper do
+
+  describe '#bootstrap_class_for' do
+    let(:flash_type) { 'alert' }
```
Comment (arun1595) [inline spec/helpers/application_helper_spec.rb]: Lets replace 'alert' with `:alert` as the key is always going to be a symbol

File: spec/helpers/application_helper_spec.rb (hunk 3 above)
Comment (KevinMulhern) [inline spec/helpers/application_helper_spec.rb]: Its not always going to be a symbol, it's a sometimes. Thats why I have `flash_type.to_sym` in the method :D using a string in the test implicitly tests that it is being converted.

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -28,4 +28,18 @@ def md(markdown_in)
     markdown = Redcarpet::Markdown.new(Redcarpet::Render::HTML, extensions = {:fenced_code_blocks => true})
     markdown.render(markdown_in).html_safe
   end
+
+  def bootstrap_class_for(flash_type)
+    bootstrap_classes.fetch(flash_type.to_sym, custom_flash(flash_type))
```
Comment (arun1595) [inline app/helpers/application_helper.rb]: Isn't `flash_type` a symbol?

Comment (KevinMulhern) [inline app/helpers/application_helper.rb]: Not all the time, its being passed in as a string in a few instances

File: spec/helpers/application_helper_spec.rb (hunk 4 above)
Comment (KevinMulhern) [inline spec/helpers/application_helper_spec.rb]: See above 😄

File: app/assets/stylesheets/static_pages.scss (hunk 1 above)
Comment (arun1595) [inline app/assets/stylesheets/static_pages.scss]: Oh, I retract my comment then.

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -5,39 +5,39 @@
     <div id="global-content-wrapper">
       <!-- turn off "global" navbar if directed to (e.g. on home page) -->
       <% if @navbar == false %>
-      
-        <%= bootstrap_flash %>
+
+        <%= render partial: 'shared/flash_helper', flash: flash %>
```
Comment (arun1595) [inline app/views/layouts/application.html.erb]: Can we replace this with `<%= render 'shared/flash_helper', flash: flash %>`?
Less keystrokes :)

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -5,39 +5,39 @@
...
         <% if flash[:partial] %>
           <%= render partial: "layouts/#{flash[:partial]}" %>
         <% end %>
-      
+
         <div class="container">
           <div class="row">
-            <%= bootstrap_flash %>
-            <%= render "shared/search_results" %>                 
+            <%= render partial: 'shared/flash_helper', flash: flash %>
```
Comment (arun1595) [inline app/views/layouts/application.html.erb]: See my comment above.

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (KevinMulhern) [inline app/helpers/application_helper.rb]: Here's why the reason for the flashes being passed in as strings 
http://blog.bigbinary.com/2014/07/24/flash-access-changes-rails-4-1.html

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -28,4 +28,18 @@ def md(markdown_in)
...
+  end
+
+  private
+
+  def custom_flash(flash_type)
+    "alert-#{flash_type.to_s}"
```
Comment (KevinMulhern) [inline app/helpers/application_helper.rb]: Good spot 👍 I'll change it now
